### PR TITLE
Fixed documentation on behaviour on return from main

### PR DESCRIPTION
**Pull request by Ioan-Cristian**
### Pull request overview

The documentation regarding the behaviour on return from main was outdated. This pull request fixes it.

### TODO

Nothing

### Documentation

+ [x] Updated relevant files.

### Format

+ [x] Run `./format_all.sh`

### CODE REVIEW

File: libtock/crt0.c [hunk 1]
```diff
@@ -160,7 +160,6 @@ void _start(void* app_start __attribute__((unused)),
     "movs r0, r6\n"             // first arg is app_start
     "movs r1, r7\n"             // second arg is mem_start
     "bl _c_start_pic\n"
-    "bkpt #255\n"
```
Comment (Ioan-Cristian) [inline libtock/crt0.c:163]: `_c_start_pic` is marked as `noreturn`, so this instruction is useless and  never reached. This change is consistent with RISC-V variant.

**Review by alistair23 — APPROVED** [review]

File: libtock/crt0.c (hunk 1 above)
Comment (ppannuto) [reply libtock/crt0.c:163]: _Technically_, a compiler _could_ ignore the `noreturn` attribute. This is a 2-byte hedge for a corner case of something going particularly wonky (see the comment above explaining its purpose). I'd rather _add_ this to RISC-V than remove it here.

**Review by ppannuto — CHANGES_REQUESTED** [review]
Looks like this got missed in #246 — thanks!

File: libtock/crt0.c (hunk 1 above)
Comment (Ioan-Cristian) [reply libtock/crt0.c:163]: This looks mostly like a defensive approach. `_c_start_pic` and `_c_start_nopic` are the functions marked as `noreturn`. Most end users will never have to touch these functions. Furthermore, modern compilers are good enough to determine if a `noreturn` function has an execution path that leads to exiting the function. The only way one could return from a `noreturn` function is through inline assembly, but at that point everything is possible, including modifying the return address to point past the break point making `bkpt` and `ebreak` useless.

**Review by ppannuto — APPROVED** [review]
I'd still rather be defensive, as I've been the recipient of the person who overrode startup symbols (but didn't say that) but then sent a bug report of things not working..., however, it's enough of a rare case that I'd rather just have the doc fix here and move on from my prior traumas 😓.

**Comment by Ioan-Cristian** [comment]
Tock doesn't use either [bkpt](https://github.com/tock/tock/blob/1d9c6412f95bd8a098b3a378a7b2f91779a3ed9b/arch/cortex-m/src/lib.rs#L191) or [ebreak](https://github.com/tock/tock/blob/1d9c6412f95bd8a098b3a378a7b2f91779a3ed9b/arch/rv32i/src/lib.rs#L138) even though in theory the main function could return through inline assembly.

**Pull request merged**